Lib/ast.py

```diff
@@ -1699,7 +1699,8 @@ def unparse(ast_obj):
 def main():
     import argparse
 
-    parser = argparse.ArgumentParser(prog='python -m ast')
+    program_name = argparse.PrettyExecutableName("ast")
+    parser = argparse.ArgumentParser(prog=f"{program_name}")
```

Comment:
With my suggestion this would become simpler. e.g.
```python
parser = argparse.ArgumentParser(prog=argparse.adjust_command('python -m ast'))
```
which is also clearer than hiding the `python -m` in the function, as you've done in your implementation.